pageserver/src/walingest.rs

```diff
@@ -1316,6 +1316,10 @@ impl WalIngest {
             }
         });
 
+        if info == pg_constants::XLOG_CHECKPOINT_SHUTDOWN {
+            modification.tline.prepare_basebackup(lsn);
+        }
```

Comment:
Would be nice if you checked the feature flag here to make it obvious instead of in the function.

Reply:
Thanks for review.
I'll merge it as is to not wait for the CI checks again, and then I will create a follow-up PR to fix this